### PR TITLE
Improve test coverage in Postgrest

### PR DESCRIPTION
**Pull request by hieuwu**
## What kind of change does this PR introduce?
Improve test coverage of Postgrest

## What is the current behavior?
There are missing tests for behavior of requests and classes in Postgrest module

## What is the new behavior?
Cover tests for missing functionalities

## Additional context
Before
Screenshot 2026-04-28 at 16 22 02

After
Screenshot 2026-04-28 at 16 34 58

### CODE REVIEW

File: Postgrest/src/commonTest/kotlin/PostgrestUpdateTest.kt [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+        val testGeneric: TestInnerObj
+    )
+
+    @Serializable
+    data class TestInnerObj(val innerMsg: String)
+
+    private val serializer = KotlinXSerializer()
+
+    @Test
+    fun testSetToGeneric() {
```
Comment (jan-tennert) [inline Postgrest/src/commonTest/kotlin/PostgrestUpdateTest.kt]: These are probably failing because they are only available on JVM/Android

Comment (hieuwu) [reply Postgrest/src/commonTest/kotlin/PostgrestUpdateTest.kt]: Let me see

Comment (hieuwu) [reply Postgrest/src/commonTest/kotlin/PostgrestUpdateTest.kt]: I change the default value of `buildPostgrestUpdate` to be `CAMEL_CASE_TO_SNAKE_CASE` at https://github.com/supabase-community/supabase-kt/pull/1275/commits/3413b966fcb789b0b29660591ec28fb678969a7c
Default value of `propertyConversionMethod` in the  `Config` also the same so it should not lead to any breaking change

File: Postgrest/src/commonTest/kotlin/PostgrestUpdateTest.kt [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+import io.github.jan.supabase.serializer.KotlinXSerializer
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class PostgrestUpdateTest {
```
Comment (jan-tennert) [inline Postgrest/src/commonTest/kotlin/PostgrestUpdateTest.kt]: Do we need that many tests for null values?

File: Postgrest/src/commonTest/kotlin/PostgrestTest.kt [hunk 3]
```diff
@@ -130,6 +184,49 @@ class PostgrestTest {
...
+                assertEquals(expectedBody, receivedBody)
+                assertEquals("data", it.url.parameters["columns"])
+                assertEquals("schema", it.headers["Content-Profile"])
+                respond("")
+            }
+        )
+    }
+
     @Test
     fun testUpsert() {
```
Comment (jan-tennert) [inline Postgrest/src/commonTest/kotlin/PostgrestTest.kt]: I'm not sure if that test is still necessary if we have those other two

File: Postgrest/src/commonTest/kotlin/PostgrestTest.kt [hunk 4]
```diff
@@ -105,6 +142,23 @@ class PostgrestTest {
...
+            requestHandler = {
+                assertMethodIs(HttpMethod.Head, it.method)
+                assertEquals(columns.value, it.url.parameters["select"])
+                respond("")
+            }
+        )
+    }
+
     @Test
     fun testInsert() {
```
Comment (jan-tennert) [inline Postgrest/src/commonTest/kotlin/PostgrestTest.kt]: same with this one

File: Postgrest/src/commonTest/kotlin/PostgrestTest.kt [hunk 5]
```diff
@@ -155,6 +252,49 @@ class PostgrestTest {
...
+                assertEquals(expectedBody, receivedBody)
+                assertEquals("data", it.url.parameters["columns"])
+                assertEquals("schema", it.headers["Content-Profile"])
+                respond("")
+            }
+        )
+    }
+
     @Test
     fun testUpdate() {
```
Comment (jan-tennert) [inline Postgrest/src/commonTest/kotlin/PostgrestTest.kt:207]: probably a few of the tests in here might be duplicates

File: Postgrest/src/commonTest/kotlin/PostgrestUpdateTest.kt (hunk 2 above)
Comment (hieuwu) [reply Postgrest/src/commonTest/kotlin/PostgrestUpdateTest.kt]: It is to cover all supported types in the request. I removed the test file at https://github.com/supabase-community/supabase-kt/pull/1275/changes/01c2b78625214225a31d9f61797a0d6f6e56862f. Seems it makes no sense just to test get, set function. The increased coverage is not worth though.

File: Postgrest/src/commonTest/kotlin/PostgrestTest.kt (hunk 5 above)
Comment (hieuwu) [reply Postgrest/src/commonTest/kotlin/PostgrestTest.kt:207]: @jan-tennert I simplified the test at https://github.com/supabase-community/supabase-kt/pull/1275/commits/5d129ecc42bbc720648128d6127a821697223266. 

This is the latest coverage. > 80% is fine now.
Screenshot 2026-05-07 at 21 20 42

Comment (hieuwu) [reply Postgrest/src/commonTest/kotlin/PostgrestTest.kt:207]: Also I've noticed that overall currently there are many ways to do the same thing like get, set and build object, which makes many public function calls to the same function. That's why when try to increase coverage for just small number of lines it leads to lots of duplicate tests. 
I'll try to analyze and find some way to improve the test structure

**Review by jan-tennert — APPROVED** [review]
